crates/pgt_statement_splitter/src/parser/common.rs

```diff
@@ -125,6 +149,36 @@ pub(crate) fn unknown(p: &mut Parser, exclude: &[SyntaxKind]) {
             } => {
                 case(p);
             }
+            Token {
+                kind: SyntaxKind::Ascii92,
+                ..
+            } => {
+                // pgsql commands e.g.
+                //
+                // ```
+                // \if test
+                // ```
+                //
+                // we wait for "\" and check if the previous token is a newline
+
+                // newline is a whitespace, but we do not want to ignore it here
+                let irrelevant = WHITESPACE_TOKENS
+                    .iter()
+                    .filter(|t| **t != SyntaxKind::Newline)
+                    .collect::<Vec<_>>();
+
+                // go back from the current position without ignoring irrelevant tokens
+                if p.tokens
+                    .iter()
+                    .take(p.current_pos)
+                    .rev()
+                    .find(|t| !irrelevant.contains(&&t.kind))
+                    .is_some_and(|t| t.kind == SyntaxKind::Newline)
```

Comment:
The `is_irrelevant` function in the parser already checks for `!= Newline`. Is that broken maybe?

Reply:
maybe `p.look_back()` just works

Reply:
Ah I see! thanks 💪🏻